Detectors/CTP/simulation/include/CTPSimulation/Digits2Raw.h

```diff
@@ -0,0 +1,78 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file Digits2Raw.h
+/// \brief Digits  tw Raw translation
+/// \author Roman Lietava
+
+#ifndef ALICEO2_CTP_DIGITS2RAW_H_
+#define ALICEO2_CTP_DIGITS2RAW_H_
+
+#include "DetectorsRaw/RawFileWriter.h"
+#include "DataFormatsCTP/Configuration.h"
+#include "DataFormatsCTP/Digits.h"
+#include "TRandom.h"
+#include "DetectorsRaw/RawFileWriter.h"
+#include "DetectorsRaw/RDHUtils.h"
+#include "DetectorsRaw/HBFUtils.h"
+#include "CCDB/BasicCCDBManager.h"
+
+
+namespace o2
+{
+namespace ctp
+{
+class Digits2Raw
+{
+ public:
+  static constexpr uint32_t NGBT=80;
+  Digits2Raw() = default;
+  ~Digits2Raw() = default;
+  void init();
+  void setCCDBServer(const std::string& server) {mCCDBServer = server;}
+  void setVerbosity(int v) { mVerbosity = v; }
+  void setFilePerLink(bool v) { mOutputPerLink = v; }
+  bool getFilePerLink() const { return mOutputPerLink; }
+  void setOutDir(const std::string& outDir){mOutDir = outDir;}
+  void processDigits(const std::string& fileDigitsName);
+  void emptyHBFMethod(const header::RDHAny* rdh, std::vector<char>& toAdd) const;
+  std::vector<char> digits2HBTPayload(const gsl::span<std::bitset<NGBT>> digits, uint32_t Npld) const; 
+  bool makeGBTWord(const std::bitset<NGBT> &pld,std::bitset<NGBT> &gbtword, uint32_t& size_gbt, uint32_t Npld) const;
+  int digit2GBTdigit(std::bitset<NGBT>& gbtdigit, const CTPDigit& digit);
+ private:  
+  // CTP configuration 
+  std::string mCCDBServer="http://ccdb-test.cern.ch:8080";  
+  CTPConfiguration* mCTPConfiguration = nullptr; 
+  //std::vector<CTPInput> *mInputs = nullptr;
+  // Raw Writer
+  o2::raw::RawFileWriter mWriter{"CTP"};
+  int mVerbosity = 0;
+  bool mOutputPerLink = false;
+  uint32_t mLinkID = 0;
+  uint16_t mCruID = 0;
+  uint32_t mEndPointID = 0;
+  uint64_t mFeeID = 0;
+  std::string mOutDir;
+  uint32_t mActiveLink = -1;
+  // CTP specific
+  const int mNLinks=2;
+  const uint32_t CRULinkIDIntRec = 0;
+  const uint32_t NIntRecPayload = 48+12;
+  bool mZeroSuppressedIntRec = false;
+  const uint32_t CRULinkIDClassRec = 1;
+  const uint32_t NClassPayload = 64+12;
+  bool mZeroSuppressedClassRec = true;
+  //constexpr uint32_t CTPCRULinkIDMisc = 2;
+  const std::string HBFEmpty = "EMPTY_HBF       ";
+  ClassDefNV(Digits2Raw, 1);
```

Comment:
this will be a simple executable, why do you need root dictionary?

Reply:
This is a class which would be used by simple executable. Do I need root dictionary ?

Reply:
no

Reply:
@shahor02  I removed Digits2Raw from LinkDef. Also there is no ClassImp in cxx. But when I remove also line ClassDefNV(Digits2Raw, 1); from header it does not compile,  so I left it. Is it ok ?

Reply:
I suspect this happens because you did not suppress https://github.com/AliceO2Group/AliceO2/blob/da03f266d029beb6119a6742a3abd18ee08aef34/Detectors/CTP/simulation/CMakeLists.txt#L20 after suppressing  `ClassDefNV(Digits2Raw, 1);`

Reply:
After removing it still complains, please, see below.
Do I need to clean ?
R.
Detectors/CTP/simulation/G__O2CTPSimulation.cxx: In function 'ROOT::TGenericClassInfo* ROOT::GenerateInitInstanceLocal(const o2::ctp::Digits2Raw*)':
Detectors/CTP/simulation/G__O2CTPSimulation.cxx:91:65: error: 'Class_Version' is not a member of 'o2::ctp::Digits2Raw'
   91 |          instance("o2::ctp::Digits2Raw", ::o2::ctp::Digits2Raw::Class_Version(), "", 134,
      |                                                                 ^~~~~~~~~~~~~
Detectors/CTP/simulation/G__O2CTPSimulation.cxx:93:43: error: 'Dictionary' is not a member of 'o2::ctp::Digits2Raw'
   93 |                   &::o2::ctp::Digits2Raw::Dictionary, isa_proxy, 4,
      |                                           ^~~~~~~~~~